### ISSUE TITLE
Problem podczas aktualizacji ofert

### ISSUE THREAD

**Issue by AureliuszBrussy** [allegro/allegro-api#10229]
### Discussed in https://github.com/allegro/allegro-api/discussions/10227

<div type='discussions-op-text'>

<sup>Originally posted by **ryszardlet** November  5, 2024</sup>
Podczas prób aktualizacji oferty (właściwie to chodzi o zmianę dostępności produktu - ilości, metoda: https://api.{environment}/sale/product-offers/{offerId}) dostaję błąd:
`"{"errors":[{"code":"PARAMETER_MISMATCH","message":"The provided parameter 'Marka'(3786) value (bez marki(3786_1704209)) does not match the existing parameter value (inna(3786_85))","details":null,"path":"productSet
[0].product.parameters","userMessage":"The specified product exists. The specified parameter `Marka` with the value `bez marki(3786_1704209)` does not match the product parameter `inna(3786_85)`.","metadata":{"productId":"XXXXXX"}}]}"`. O ile potrafiłbym zrozumieć błąd, który wynikałbym z niewłaściwej konfiguracji obsługi o tyle jest to dla mnie niejasne, że wcześnie (przed aktualizacją) pobieram ofertę - modyfikuję co potrzebuję i odsyłam. A w odpowiedzi na zapytanie o ofertę mam:
` values: [bez marki]
  valuesIds: [3786_1704209]` 
  jak zatem z tego wybrnąć? </div>

**Comment by AureliuszBrussy** [comment]
@ryszardlet przyślij proszę trace-id Twojego wywołania, które kończy się tym błędem. Trace-id znajdziesz w nagłówku naszego response'a. Dzięki temu będę mógł dokładnie sprawdzić ten przypadek.

**Comment by ryszardlet** [comment]
Niestety nie mam łatwego sposobu dostać się do tych danych. Czy w API w ostatnim czasie się coś zmieniło? Na tyle znacząco, że może to nie działać? Tak na marginesie - przewiduja państwo może metodę do aktualizacji stanów?

**Comment by AureliuszBrussy** [comment]
Nie, nie zmienialiśmy ostatnio nic w endpointach do zarządzania ofertą. 

Możesz aktualizować stany korzystając z dedykowanego endpointu: [PUT /sale/offer-quantity-change-commands/{commandId}](https://developer.allegro.pl/documentation#operation/quantityModificationCommandUsingPUT).

Dodam też, że korzystac z [PATCH /sale/product-offers/{offerId}](https://developer.allegro.pl/documentation/#operation/editProductOffers) możesz wysyłać tylko zmianę liczby sztuk. Przykład znajdziesz [w naszym poradniku](https://developer.allegro.pl/tutorials/jak-zarzadzac-ofertami-7GzB2L37ase#edycja-pol-oferty).

**Comment by ryszardlet** [comment]
Dopytam jeszcze - w jaki sposób zarządzają państwo zmianami w API? Np. teraz zorientowałem się, że zniknęła sekcja tax (i obecnie API zwraca błąd). Jak często tego rodzaju zmiany są wprowadzane? Jak mogę jako developer obserwować te zmiany i móc na nie reagować?

**Comment by AureliuszBrussy** [comment]
Najlepiej, jeżeli [zasubskrybujesz newsletter devportalu](https://developer.allegro.pl/newsletter). Otrzymasz wszelkie informacje o zmianach na bieżąco. Warto także co jakiś czas spojrzeć na sekcję [zaplanowane zmiany](https://developer.allegro.pl/news) na stronie z naszym newsami. Tam listujemy najbardziej istotne, nadchodzące zmiany wraz datą ich wdrożenia.

**Comment by ryszardlet** [comment]
Ok. Zapisany.
